loadbalancer/algorithm_test.go

```diff
@@ -312,6 +312,58 @@ func TestConsistentHashSearch(t *testing.T) {
 	}
 }
 
+func TestConsistentHashBoundedLoadSearch(t *testing.T) {
+	endpoints := []string{"http://127.0.0.1:8080", "http://127.0.0.2:8080", "http://127.0.0.3:8080"}
+	ch := newConsistentHash(endpoints).(consistentHash)
+
+	r, _ := http.NewRequest("GET", "http://127.0.0.1:1234/foo", nil)
+	r.RemoteAddr = "192.168.0.1:8765"
+	route := NewAlgorithmProvider().Do([]*routing.Route{{
+		Route: eskip.Route{
+			BackendType: eskip.LBBackend,
+			LBAlgorithm: ConsistentHash.String(),
+			LBEndpoints: endpoints,
+		},
+	}})[0]
+	ctx := &routing.LBContext{Request: r, Route: route, Params: map[string]interface{}{ConsistentHashKey: net.RemoteHost(r).String()}}
+	noLoad := ch.boundedLoadSearch("somekey", 1.25, ctx)
```

Comment:
Should we rather test using `ch.Apply(&routing.LBContext{Request: r, Route: rt, Params: map[string]interface{...}})`?

To compare bounded with non bounded we may call once with `ConsistentHashBalanceFactor: 1.25` param and once without.

Reply:
Maybe drop `ConsistentHashKey` param for clarity - it is not relevant to the bounded algo

Reply:
Yeah true I'm missing a new if branch if I don't test Apply, I'll switch over to that